nvme.c

```diff
@@ -5317,6 +5321,330 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+		 OPT_FLAG("bprsel",         'B', &cfg.bprsel,         bprsel),
+		 OPT_FLAG("bpmbl",          'p', &cfg.bpmbl,          bpmbl),
+		 OPT_FLAG("cmbmsc",         's', &cfg.cmbmsc,         cmbmsc),
+		 OPT_FLAG("cmbsts",         'S', &cfg.cmbsts,         cmbsts),
+		 OPT_FLAG("crto",           'r', &cfg.crto,           crto),
+		 OPT_FLAG("pmrcap",         'P', &cfg.pmrcap,         pmrcap),
+		 OPT_FLAG("pmrctl",         'R', &cfg.pmrctl,         pmrctl),
+		 OPT_FLAG("pmrsts",         't', &cfg.pmrsts,         pmrsts),
+		 OPT_FLAG("pmrebs",         'e', &cfg.pmrebs,         pmrebs),
+		 OPT_FLAG("pmrswtp",        'w', &cfg.pmrswtp,        pmrswtp),
+		 OPT_FLAG("pmrmscl",        'l', &cfg.pmrmscl,        pmrmscl),
+		 OPT_FLAG("pmrmscu",        'u', &cfg.pmrmscu,        pmrmscu));
+
+	err = parse_and_open(&dev, argc, argv, desc, opts);
+	if (err)
+		return err;
+
+	if (!argconfig_parse_seen(opts, "offset")) {
+		if (cfg.cap)
+			cfg.offset = NVME_REG_CAP;
+		else if (cfg.vs)
+			cfg.offset = NVME_REG_VS;
+		else if (cfg.intms)
+			cfg.offset = NVME_REG_INTMS;
+		else if (cfg.intmc)
+			cfg.offset = NVME_REG_INTMC;
+		else if (cfg.cc)
+			cfg.offset = NVME_REG_CC;
+		else if (cfg.csts)
+			cfg.offset = NVME_REG_CSTS;
+		else if (cfg.nssr)
+			cfg.offset = NVME_REG_NSSR;
+		else if (cfg.aqa)
+			cfg.offset = NVME_REG_AQA;
+		else if (cfg.asq)
+			cfg.offset = NVME_REG_ASQ;
+		else if (cfg.acq)
+			cfg.offset = NVME_REG_ACQ;
+		else if (cfg.cmbloc)
+			cfg.offset = NVME_REG_CMBLOC;
+		else if (cfg.cmbsz)
+			cfg.offset = NVME_REG_CMBSZ;
+		else if (cfg.bpinfo)
+			cfg.offset = NVME_REG_BPINFO;
+		else if (cfg.bprsel)
+			cfg.offset = NVME_REG_BPRSEL;
+		else if (cfg.bpmbl)
+			cfg.offset = NVME_REG_BPMBL;
+		else if (cfg.cmbmsc)
+			cfg.offset = NVME_REG_CMBMSC;
+		else if (cfg.cmbsts)
+			cfg.offset = NVME_REG_CMBSTS;
+		else if (cfg.crto)
+			cfg.offset = NVME_REG_CRTO;
+		else if (cfg.pmrcap)
+			cfg.offset = NVME_REG_PMRCAP;
+		else if (cfg.pmrctl)
+			cfg.offset = NVME_REG_PMRCTL;
+		else if (cfg.pmrsts)
+			cfg.offset = NVME_REG_PMRSTS;
+		else if (cfg.pmrebs)
+			cfg.offset = NVME_REG_PMREBS;
+		else if (cfg.pmrswtp)
+			cfg.offset = NVME_REG_PMRSWTP;
+		else if (cfg.pmrmscl)
+			cfg.offset = NVME_REG_PMRMSCL;
+		else if (cfg.pmrmscu)
+			cfg.offset = NVME_REG_PMRMSCU;
+	}
+
+	if (cfg.offset < 0) {
+		nvme_show_error("offset required param");
+		return -EINVAL;
+	}
+
+	err = validate_output_format(output_format_val, &flags);
+	if (err < 0) {
+		nvme_show_error("Invalid output format");
+		return err;
+	}
+
+	if (cfg.human_readable)
+		flags |= VERBOSE;
+
+	r = nvme_scan(NULL);
+	bar = mmap_registers(r, dev, false);
+	if (!bar) {
+		err = get_register_property(dev_fd(dev), &bar, cfg.offset);
+		if (err)
+			return err;
+		fabrics = true;
+	}
+
+	nvme_show_ctrl_register(bar, fabrics, cfg.offset, flags);
+	if (fabrics)
+		free(bar);
+	else
+		munmap(bar, getpagesize());
+
+	return err;
```

Comment:
Could just `return 0` here too

Reply:
Fixed set-reg command also as same as mentioned by the commit 1842cbc0.